testing/test_collection.py

```diff
@@ -1157,3 +1157,20 @@ def test_collectignore_via_conftest(testdir, monkeypatch):
 
     result = testdir.runpytest()
     assert result.ret == EXIT_NOTESTSCOLLECTED
+
+
+def test_collect_pkg_init_and_file_in_args(testdir):
```

Comment:
Hmm this test is passing for me without your patch (`master`). Can you double check that this is reproducing the problem for you please?

Reply:
Oh, snap - thanks for catching.

The difference/fix is only without `-o python_files=*.py` then - extended the test.
Not sure if this is really expected, but better then not collection anything I guess.

Reply:
I see, thanks. I checked and the new test does fail with `master`. 👍